embulk-standards/src/main/java/org/embulk/standards/JsonParserPlugin.java

```diff
@@ -99,12 +131,79 @@ private PageBuilder newPageBuilder(Schema schema, PageOutput output)
         return new PageBuilder(Exec.getBufferAllocator(), schema, output);
     }
 
-    private JsonParser.Stream newJsonStream(FileInputInputStream in)
-            throws IOException
-    {
-        return new JsonParser().open(in);
+    private JsonParser.Stream newJsonStream(FileInputInputStream in , PluginTask task)
+            throws IOException {
+        InvalidEscapeStringPolicy policy = task.getInvalidEscapeStringPolicy();
+        switch (policy) {
+            case SKIP:
+            case UNESCAPE:
+                Iterable<CharSource> lines = Lists.transform(CharStreams.readLines(new BufferedReader(new InputStreamReader(in))),
+                        invalidEscapeStringFunction(policy));
+                return new JsonParser().open(new ReaderInputStream(CharSource.concat(lines).openStream()));
+            case PASSTHROUGH:
+            default:
+                return new JsonParser().open(in);
+        }
+    }
+
+    Function<String, CharSource> invalidEscapeStringFunction(final InvalidEscapeStringPolicy policy) {
+        return new Function<String, CharSource>() {
+            final Pattern p = Pattern.compile("\\p{XDigit}+");
```

Comment:
Let's use a more descriptive variable name -- such as `digitsPattern` or like that.